### PR TITLE
UI tests for adminSharingSettings

### PR DESCRIPTION
**Pull request by paurakhsharma**
## Description
UI test for testing different admin sharing settings

## Related Issue
#31983

## How Has This Been Tested?
Locally

## Types of changes
- [x] Tests

## Checklist:
- [x] Acceptance tests added

### CODE REVIEW

File: tests/acceptance/features/bootstrap/WebUIAdminSharingSettingsContext.php [hunk 1]
```diff
@@ -0,0 +1,217 @@
...
+	 * WebUIAdminSharingSettingsContext constructor.
+	 *
+	 * @param AdminSharingSettingsPage $adminSharingSettingsPage
+	 * @param LoginPage $loginPage
+	 */
+	public function __construct(
+		AdminSharingSettingsPage $adminSharingSettingsPage,
+		LoginPage $loginPage
+	) {
+			$this->adminSharingSettingsPage = $adminSharingSettingsPage;
```
Comment (phil-davis) [inline tests/acceptance/features/bootstrap/WebUIAdminSharingSettingsContext.php]: indent

File: tests/acceptance/features/bootstrap/WebUIAdminSharingSettingsContext.php [hunk 2]
```diff
@@ -0,0 +1,217 @@
...
+	 */
+	public function theAdminEnforcesPasswordProtectionForReadAndWriteLinksUsingTheWebui() {
+		$this->adminSharingSettingsPage->enforcePasswordProtectionForReadWriteLinks();
+		$this->adminSharingSettingsPage->waitForAjaxCallsToStartAndFinish(
+			$this->getSession()
+		);
+	}
+
+	/**
+	 * @When the admin enforces password protection upload only links using the webUI
```
Comment (phil-davis) [inline tests/acceptance/features/bootstrap/WebUIAdminSharingSettingsContext.php]: for upload only

File: tests/acceptance/features/bootstrap/WebUIAdminSharingSettingsContext.php [hunk 3]
```diff
@@ -0,0 +1,217 @@
...
+	 */
+	public function theAdminDisablesResharingUsingTheWebui() {
+		$this->adminSharingSettingsPage->disableResharing();
+		$this->adminSharingSettingsPage->waitForAjaxCallsToStartAndFinish(
+			$this->getSession()
+		);
+	}
+
+	/**
+	 * @When the admin disables sharing with group using the webUI
```
Comment (phil-davis) [inline tests/acceptance/features/bootstrap/WebUIAdminSharingSettingsContext.php]: groups

File: tests/acceptance/features/bootstrap/WebUIAdminSharingSettingsContext.php [hunk 4]
```diff
@@ -0,0 +1,217 @@
...
+	 */
+	public function theAdminDisablesSharingWithGroupUsingTheWebui() {
+		$this->adminSharingSettingsPage->disableGroupSharing();
+		$this->adminSharingSettingsPage->waitForAjaxCallsToStartAndFinish(
+			$this->getSession()
+		);
+	}
+
+	/**
+	 * @When the admin restricts users to only share with their group member using the webUI
```
Comment (phil-davis) [inline tests/acceptance/features/bootstrap/WebUIAdminSharingSettingsContext.php]: members

File: tests/acceptance/features/lib/AdminSharingSettingsPage.php [hunk 5]
```diff
@@ -0,0 +1,196 @@
...
+	protected $shareFileViaSocialMediaOnPublicShareCheckboxXpath = '//label[@for="allowSocialShare"]';
+	protected $enforceLinkPasswordReadOnlyCheckboxXpath = '//label[@for="enforceLinkPasswordReadOnly"]';
+	protected $enforceLinkPasswordReadWriteCheckboxXpath = '//label[@for="enforceLinkPasswordReadWrite"]';
+	protected $enforceLinkPasswordWriteOnlyCheckboxXpath = '//label[@for="enforceLinkPasswordWriteOnly"]';
+	protected $allowResharingCheckboxXpah = '//label[@for="allowResharing"]';
+	protected $allowGroupSharingCheckboxXpath = '//label[@for="allowGroupSharing"]';
+	protected $onlyShareWithGroupMembersCheckboxXpath = '//label[@for="onlyShareWithGroupMembers"]';
+
+	/**
+	 * Disable apps to use the Share API
```
Comment (phil-davis) [inline tests/acceptance/features/lib/AdminSharingSettingsPage.php]: Disable the Share API

File: tests/acceptance/features/webUIAdminSettings/adminSharingSetting.feature [hunk 6]
```diff
@@ -0,0 +1,85 @@
+@webUI
+Feature: admin sharing settings
```
Comment (phil-davis) [inline tests/acceptance/features/webUIAdminSettings/adminSharingSetting.feature]: filename also should be Settings

File: tests/acceptance/features/webUIAdminSettings/adminSharingSetting.feature [hunk 7]
```diff
@@ -0,0 +1,85 @@
...
+As a admin
```
Comment (phil-davis) [inline tests/acceptance/features/webUIAdminSettings/adminSharingSetting.feature]: Indent
As an admin

File: tests/acceptance/features/webUIAdminSettings/adminSharingSetting.feature [hunk 8]
```diff
@@ -0,0 +1,85 @@
...
+I want to be able to manage sharing settings ownCloud server
```
Comment (phil-davis) [inline tests/acceptance/features/webUIAdminSettings/adminSharingSetting.feature]: on the ownCloud server

File: tests/acceptance/features/webUIAdminSettings/adminSharingSetting.feature [hunk 9]
```diff
@@ -0,0 +1,85 @@
...
+So that I can enable, disable, allow or restrict different sharing behaviour
+
+	Background:
+			Given the admin has browsed to the admin sharing settings page
```
Comment (phil-davis) [inline tests/acceptance/features/webUIAdminSettings/adminSharingSetting.feature]: indent

File: tests/acceptance/features/webUIAdminSettings/adminSharingSetting.feature [hunk 10]
```diff
@@ -0,0 +1,85 @@
...
+				| files_sharing | public@@@send_mail | 1     |
+
+	Scenario: disable social media share on public share
+			When the admin disables social media share on public share using the webUI
+			And the user retrieves the capabilities using the API
+			Then the capabilities should contain
+				| capability    | path_to_element       | value |
+				| files_sharing | public@@@social_share | EMPTY |
+
+	Scenario: Enforce password protection for read-only links
```
Comment (phil-davis) [inline tests/acceptance/features/webUIAdminSettings/adminSharingSetting.feature]: enforce - start with lowercase...

**Review by phil-davis — COMMENTED** [review]
See comments

**Comment by paurakhsharma** [comment]
Yeah, we have to test those scenarios too.

**Review by phil-davis — APPROVED** [review]

**Comment by paurakhsharma** [comment]
Backport on #32017

**Pull request merged**